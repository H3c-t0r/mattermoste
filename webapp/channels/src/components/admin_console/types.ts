// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {MessageDescriptor} from 'react-intl';

import type {CloudState, Product} from '@mattermost/types/cloud';
import type {AdminConfig, ClientLicense} from '@mattermost/types/config';
import type {JobType} from '@mattermost/types/jobs';

import type Constants from 'utils/constants';

import type ValidationResult from './validation';

type Component = any

type AdminDefinitionConfigSchemaComponent = {
    id: string;
    component: Component;
}

export type ConsoleAccess = {read: {[key: string]: boolean}; write: {[key: string]: boolean}}

type Validator = (value: any) => ValidationResult

type AdminDefinitionSettingCustom = Omit<AdminDefinitionSettingBase, 'label'> & {
    type: 'custom';
    key: string;
    showTitle?: boolean;
    component: Component;
    label?: string;
}

type AdminDefinitionSettingBase = {
    key?: string;
    label: MessageDescriptor;
    label_values?: {[key: string]: any};
    help_text?: string | JSX.Element | MessageDescriptor;
    help_text_markdown?: boolean;
    help_text_values?: {[key: string]: any};
    disabled_help_text?: string | JSX.Element | MessageDescriptor;
    disabled_help_text_markdown?: boolean;
    disabled_help_text_values?: {[key: string]: any};
    onConfigLoad?: (configVal: any, config: Partial<AdminConfig>) => any;
    onConfigSave?: (displayVal: any, previousVal?: any) => any;
    isHidden?: Check;
    isDisabled?: Check;
}

export type AdminDefinitionSettingBanner = AdminDefinitionSettingBase & {
    type: 'banner';
    label_markdown?: boolean;
    banner_type: 'info' | 'warning';
}

type AdminDefinitionSettingInput = AdminDefinitionSettingBase & {
    type: 'text' | 'bool' | 'longtext' | 'number' | 'color';
    placeholder?: MessageDescriptor;
    validate?: Validator;
    setFromMetadataField?: string;
    dynamic_value?: (value: any, config: Partial<AdminConfig>, state: any) => string;
    max_length?: number;
}

type AdminDefinitionSettingGenerated = AdminDefinitionSettingBase & {
    type: 'generated';
    placeholder?: string;
    regenerate_help_text?: string;
}

export type AdminDefinitionSettingDropdownOption = {
    value: string;
    display_name: MessageDescriptor;
    help_text?: MessageDescriptor;
    help_text_markdown?: boolean;
    help_text_values?: {[key: string]: any};
    isHidden?: Check;
}

type AdminDefinitionSettingDropdown = AdminDefinitionSettingBase & {
    type: 'dropdown';
    options: AdminDefinitionSettingDropdownOption[];
    isHelpHidden?: Check;
}

type AdminDefinitionSettingFileUpload = AdminDefinitionSettingBase & {
    type: 'fileupload';
    remove_help_text: MessageDescriptor;
    remove_button_text: MessageDescriptor;
    removing_text: MessageDescriptor;
    uploading_text: MessageDescriptor;
    fileType: string;
    upload_action: (file: File, success: (data: any) => void, error: (err: any) => void) => void;
    set_action?: () => void;
    setFromMetadataField?: string;
    remove_action: (success: (data: any) => void, error: (err: any) => void) => void;
}

type AdminDefinitionSettingJobsTable = AdminDefinitionSettingBase & {
    type: 'jobstable';
    job_type: JobType;
    render_job: Component;
};

type AdminDefinitionSettingLanguage = AdminDefinitionSettingBase & {
    type: 'language';
    multiple?: boolean;
    no_result?: MessageDescriptor;
}

type AdminDefinitionSettingButton = AdminDefinitionSettingBase & {
    type: 'button';
    action: (success: (data?: any) => void, error: (error: {message: string; detailed_error?: string}) => void, siteUrl: string) => void;
    loading?: MessageDescriptor;
    error_message: MessageDescriptor;
    success_message?: MessageDescriptor;
    sourceUrlKey?: string;
}

type AdminDefinitionSettingUsername = AdminDefinitionSettingBase & {
    type: typeof Constants.SettingsTypes.TYPE_USERNAME;
    placeholder_message: string;
}

type AdminDefinitionSettingPermission = AdminDefinitionSettingBase & {
    type: typeof Constants.SettingsTypes.TYPE_PERMISSION;
}

type AdminDefinitionSettingRadio = AdminDefinitionSettingBase & {
    type: typeof Constants.SettingsTypes.TYPE_RADIO;
    options: AdminDefinitionSettingDropdownOption[];
}

export type AdminDefinitionSetting = AdminDefinitionSettingCustom |
AdminDefinitionSettingInput | AdminDefinitionSettingGenerated |
AdminDefinitionSettingBanner | AdminDefinitionSettingDropdown |
AdminDefinitionSettingButton | AdminDefinitionSettingFileUpload |
AdminDefinitionSettingJobsTable | AdminDefinitionSettingLanguage |
AdminDefinitionSettingUsername | AdminDefinitionSettingPermission |
AdminDefinitionSettingRadio;

type AdminDefinitionConfigSchemaSettings = {
    id: string;
    name: string | MessageDescriptor;
    isHidden?: Check;
    onConfigLoad?: (config: Partial<AdminConfig>) => {[x: string]: string};
    onConfigSave?: (displayVal: any) => any;
    settings?: AdminDefinitionSetting[];
    sections?: AdminDefinitionConfigSchemaSection[];
    footer?: string;
    header?: string;
}

type AdminDefinitionConfigSchemaSection = {
    title: string;
    subtitle?: string;
    settings: AdminDefinitionSetting[];
    header?: string;
    footer?: string;
}

type RestrictedIndicatorType = {
    value: (cloud: CloudState) => JSX.Element;
    shouldDisplay: (license: ClientLicense, subscriptionProduct: Product|undefined) => boolean;
}

export type AdminDefinitionSubSectionSchema = AdminDefinitionConfigSchemaComponent | AdminDefinitionConfigSchemaSettings;

export type AdminDefinitionSubSection = {
    url: string;
    title?: MessageDescriptor;
    searchableStrings?: Array<string|[string, {[key: string]: any}]>;
    isHidden?: Check;
    isDiscovery?: boolean;
    isDisabled?: Check;
    schema: AdminDefinitionSubSectionSchema;
    restrictedIndicator?: RestrictedIndicatorType;
}

export type AdminDefinitionSection = {
    icon: JSX.Element;
    sectionTitle: MessageDescriptor;
    isHidden: Check;
    id?: string;
    subsections: {[key: string]: AdminDefinitionSubSection};
}

export type AdminDefinition = {[key: string]: AdminDefinitionSection}

export type Check = boolean | ((config: Partial<AdminConfig>, state: any, license?: ClientLicense, enterpriseReady?: boolean, consoleAccess?: ConsoleAccess, cloud?: CloudState, isSystemAdmin?: boolean) => boolean)
