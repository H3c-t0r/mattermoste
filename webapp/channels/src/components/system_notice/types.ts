// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {AnalyticsRow} from '@mattermost/types/admin';
import {Channel} from '@mattermost/types/channels';

export type Notice = {
    name: string;
    adminOnly?: boolean;
    title: React.ReactNode;
    icon?: React.ReactNode;
    body: React.ReactNode;
    allowForget: boolean;
    show?(
        serverVersion: string,
        config: any,
        license: any,
        analytics?: Record<string, number | AnalyticsRow[]>,
        currentChannel?: Channel,
    ): boolean;
}
