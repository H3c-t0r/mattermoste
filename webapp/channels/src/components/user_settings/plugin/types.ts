// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type PluginConfiguration = {
    id: string;
    uiName: string;
    icon?: string;
    settings: PluginConfigurationSetting[];
}

type BasePluginConfigurationSetting = {
    name: string;
    title: string;
    helpText?: string;
    onSubmit?: (name: string, value: string) => void;
    default: string;
}

type PluginConfigurationRadioSetting = BasePluginConfigurationSetting & {
    type: 'radio';
    options: Array<{
        value: string;
        text: string;
    }>;
}

export type PluginConfigurationSetting = PluginConfigurationRadioSetting
