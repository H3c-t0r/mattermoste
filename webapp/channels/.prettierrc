{
  "singleQuote": true,
  "trailingComma": "all",
  "semi": false,
  "object-curly-spacing": ["error", "never"]
}
