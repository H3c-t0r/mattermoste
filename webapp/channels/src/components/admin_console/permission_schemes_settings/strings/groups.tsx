// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {MessageDescriptor} from 'react-intl';
import {defineMessages} from 'react-intl';

export const groupRolesStrings: Record<string, Record<string, MessageDescriptor>> = {
    delete_posts: defineMessages({
        name: {
            id: 'admin.permissions.group.delete_posts.name',
            defaultMessage: 'Delete Posts',
        },
        description: {
            id: 'admin.permissions.group.delete_posts.description',
            defaultMessage: 'Delete own and others\' posts.',
        },
    }),
    integrations: defineMessages({
        name: {
            id: 'admin.permissions.group.integrations.name',
            defaultMessage: 'Integrations & Customizations',
        },
        description: {
            id: 'admin.permissions.group.integrations.description',
            defaultMessage: 'Manage OAuth 2.0, slash commands, webhooks and emoji.',
        },
    }),
    posts: defineMessages({
        name: {
            id: 'admin.permissions.group.posts.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.posts.description',
            defaultMessage: '',
        },
    }),
    private_channel: defineMessages({
        name: {
            id: 'admin.permissions.group.private_channel.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.private_channel.description',
            defaultMessage: '',
        },
    }),
    public_channel: defineMessages({
        name: {
            id: 'admin.permissions.group.public_channel.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.public_channel.description',
            defaultMessage: '',
        },
    }),
    reactions: defineMessages({
        name: {
            id: 'admin.permissions.group.reactions.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.reactions.description',
            defaultMessage: '',
        },
    }),
    send_invites: defineMessages({
        name: {
            id: 'admin.permissions.group.send_invites.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.send_invites.description',
            defaultMessage: '',
        },
    }),
    teams: defineMessages({
        name: {
            id: 'admin.permissions.group.teams.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.teams.description',
            defaultMessage: '',
        },
    }),
    edit_posts: defineMessages({
        name: {
            id: 'admin.permissions.group.edit_posts.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.edit_posts.description',
            defaultMessage: '',
        },
    }),
    teams_team_scope: defineMessages({
        name: {
            id: 'admin.permissions.group.teams_team_scope.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.teams_team_scope.description',
            defaultMessage: '',
        },
    }),
    guest_reactions: defineMessages({
        name: {
            id: 'admin.permissions.group.guest_reactions.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.guest_reactions.description',
            defaultMessage: '',
        },
    }),
    guest_create_post: defineMessages({
        name: {
            id: 'admin.permissions.group.guest_create_post.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.guest_create_post.description',
            defaultMessage: '',
        },
    }),
    guest_create_private_channel: defineMessages({
        name: {
            id: 'admin.permissions.group.guest_create_private_channel.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.guest_create_private_channel.description',
            defaultMessage: '',
        },
    }),
    guest_delete_post: defineMessages({
        name: {
            id: 'admin.permissions.group.guest_delete_post.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.guest_delete_post.description',
            defaultMessage: '',
        },
    }),
    guest_edit_post: defineMessages({
        name: {
            id: 'admin.permissions.group.guest_edit_post.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.guest_edit_post.description',
            defaultMessage: '',
        },
    }),
    guest_use_channel_mentions: defineMessages({
        name: {
            id: 'admin.permissions.group.guest_use_channel_mentions.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.guest_use_channel_mentions.description',
            defaultMessage: '',
        },
    }),
    guest_use_group_mentions: defineMessages({
        name: {
            id: 'admin.permissions.group.guest_use_group_mentions.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.guest_use_group_mentions.description',
            defaultMessage: '',
        },
    }),
    manage_private_channel_members_and_read_groups: defineMessages({
        name: {
            id: 'admin.permissions.group.manage_private_channel_members_and_read_groups.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.manage_private_channel_members_and_read_groups.description',
            defaultMessage: '',
        },
    }),
    manage_public_channel_members_and_read_groups: defineMessages({
        name: {
            id: 'admin.permissions.group.manage_public_channel_members_and_read_groups.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.manage_public_channel_members_and_read_groups.description',
            defaultMessage: '',
        },
    }),
    convert_public_channel_to_private: defineMessages({
        name: {
            id: 'admin.permissions.group.convert_public_channel_to_private.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.convert_public_channel_to_private.description',
            defaultMessage: '',
        },
    }),
    manage_shared_channels: defineMessages({
        name: {
            id: 'admin.permissions.group.manage_shared_channels.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.manage_shared_channels.description',
            defaultMessage: '',
        },
    }),
    playbook_public: defineMessages({
        name: {
            id: 'admin.permissions.group.playbook_public.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.playbook_public.description',
            defaultMessage: '',
        },
    }),
    playbook_private: defineMessages({
        name: {
            id: 'admin.permissions.group.playbook_private.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.playbook_private.description',
            defaultMessage: '',
        },
    }),
    runs: defineMessages({
        name: {
            id: 'admin.permissions.group.runs.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.runs.description',
            defaultMessage: '',
        },
    }),
    custom_groups: defineMessages({
        name: {
            id: 'admin.permissions.group.custom_groups.name',
            defaultMessage: '',
        },
        description: {
            id: 'admin.permissions.group.custom_groups.description',
            defaultMessage: '',
        },
    }),
};
