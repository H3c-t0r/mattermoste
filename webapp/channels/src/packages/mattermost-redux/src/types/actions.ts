// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {AnyAction} from 'redux';
import type {ThunkAction} from 'redux-thunk';

import type {GlobalState} from '@mattermost/types/store';

export type Thunk<ReturnType, State = GlobalState> = ThunkAction<ReturnType, State, never, AnyAction>;

export type ActionFunc<Data = any, Error = any, State = GlobalState> = Thunk<Promise<ActionResult<Data, Error>> | ActionResult<Data, Error>, State>;
export type ActionFuncPromiseDoNotUse<Data = any, Error = any> = Thunk<Promise<ActionResult<Data, Error>>>;

export type DispatchFunc<ReturnType = any, State = GlobalState> = Parameters<Thunk<ReturnType, State>>[0];
export type GetStateFunc<State = GlobalState> = Parameters<Thunk<any, State>>[1];

export type GenericAction = AnyAction;

export type ActionResult<Data = any, Error = any> = {
    data?: Data;
    error?: Error;
};
