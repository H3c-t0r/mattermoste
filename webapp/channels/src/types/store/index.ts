// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {PluginsState} from './plugins';
import type {ViewsState} from './views';
import type {GlobalState as BaseGlobalState} from '@mattermost/types/store';

export type DraggingState = {
    state?: string;
    type?: string;
    id?: string;
}

export type GlobalState = BaseGlobalState & {
    plugins: PluginsState;
    storage: {
        storage: Record<string, any>;
        initialized: boolean;
    };
    views: ViewsState;
};
