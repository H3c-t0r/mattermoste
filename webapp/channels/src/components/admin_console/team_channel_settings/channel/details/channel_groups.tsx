// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {FormattedMessage} from 'react-intl';

import {Channel} from '@mattermost/types/channels';
import {Group} from '@mattermost/types/groups';

import AddGroupsToChannelModal from 'components/add_groups_to_channel_modal';
import ToggleModalButton from 'components/toggle_modal_button';
import AdminPanel from 'components/widgets/admin_console/admin_panel';

import {ModalIdentifiers} from 'utils/constants';
import {t} from 'utils/i18n';

import GroupList from '../../group';

interface ChannelGroupsProps {
    synced: boolean;
    channel: Partial<Channel>;
    onAddCallback: (groupIDs: string[]) => void;
    totalGroups: number;
    groups: Array<Partial<Group>>;
    removedGroups: Array<{[key: string]: any}>;
    onGroupRemoved: (gid: string) => void;
    setNewGroupRole: (gid: string) => void;
    isDisabled?: boolean;
}

export const ChannelGroups: React.FunctionComponent<ChannelGroupsProps> = (props: ChannelGroupsProps): JSX.Element => {
    const {onGroupRemoved, onAddCallback, totalGroups, groups, removedGroups, channel, synced, setNewGroupRole, isDisabled} = props;
    const groupsItems: Group[] = props.groups.map((groupItem) => ({
        id: groupItem.id || '',
        name: groupItem.name || '',
        display_name: groupItem.display_name || '',
        description: groupItem.description || '',
        source: groupItem.source || '',
        remote_id: groupItem.remote_id || null,
        create_at: groupItem.create_at || -1,
        update_at: groupItem.update_at || -1,
        delete_at: groupItem.delete_at || -1,
        has_syncables: groupItem.has_syncables || false,
        member_count: groupItem.member_count || -1,
        scheme_admin: groupItem.scheme_admin || false,
        allow_reference: groupItem.allow_reference || false,
    }));
    return (
        <AdminPanel
            id='channel_groups'
            titleId={synced ? t('admin.channel_settings.channel_detail.syncedGroupsTitle') : t('admin.channel_settings.channel_detail.groupsTitle')}
            titleDefault={synced ? 'Synced Groups' : 'Groups'}
            subtitleId={synced ? t('admin.channel_settings.channel_detail.syncedGroupsDescription') : t('admin.channel_settings.channel_detail.groupsDescription')}
            subtitleDefault={synced ? 'Add and remove channel members based on their group membership.' : 'Select groups to be added to this channel.'}
            button={
                <ToggleModalButton
                    id='addGroupsToChannelToggle'
                    className='btn btn-primary'
                    modalId={ModalIdentifiers.ADD_GROUPS_TO_CHANNEL}
                    dialogType={AddGroupsToChannelModal}
                    dialogProps={{
                        channel,
                        onAddCallback,
                        skipCommit: true,
                        includeGroups: removedGroups,
                        excludeGroups: groups,
                    }}
                    disabled={isDisabled}
                >
                    <FormattedMessage
                        id='admin.channel_settings.channel_details.add_group'
                        defaultMessage='Add Group'
                    />
                </ToggleModalButton>
            }
        >
            {channel.id && (
                <GroupList
                    channel={channel}
                    groups={groupsItems}
                    totalGroups={totalGroups}
                    onGroupRemoved={onGroupRemoved}
                    setNewGroupRole={setNewGroupRole}
                    isModeSync={synced}
                    type='channel'
                    isDisabled={isDisabled}
                />
            )}
        </AdminPanel>
    );
};
