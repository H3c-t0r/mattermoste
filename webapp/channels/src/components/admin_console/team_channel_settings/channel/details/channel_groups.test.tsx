// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {Group} from '@mattermost/types/groups';
import {Channel} from '@mattermost/types/channels';

import {ChannelGroups} from './channel_groups';

describe('admin_console/team_channel_settings/channel/ChannelGroups', () => {
    test('should match snapshot', () => {
        const groups: Group[] = [{
            id: '123',
            name: '',
            display_name: 'DN',
            member_count: 3,
            description: '',
            source: '',
            remote_id: null,
            create_at: -1,
            update_at: -1,
            delete_at: -1,
            has_syncables: false,
            scheme_admin: false,
            allow_reference: false,
        }];

        const testChannel: Partial<Channel> & {team_name: string} = {
            id: '123',
            team_name: 'team',
            type: 'O',
            group_constrained: false,
            name: 'DN',
        };
        const wrapper = shallow(
            <ChannelGroups
                synced={true}
                onAddCallback={jest.fn()}
                onGroupRemoved={jest.fn()}
                removedGroups={[]}
                groups={groups}
                channel={testChannel}
                totalGroups={1}
                setNewGroupRole={jest.fn()}
                isDisabled={false}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
