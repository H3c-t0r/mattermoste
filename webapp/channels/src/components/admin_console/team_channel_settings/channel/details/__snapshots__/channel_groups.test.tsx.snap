// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin_console/team_channel_settings/channel/ChannelGroups should match snapshot 1`] = `
<AdminPanel
  button={
    <Memo(Connect(ToggleModalButton))
      className="btn btn-primary"
      dialogProps={
        Object {
          "channel": Object {
            "group_constrained": false,
            "id": "123",
            "name": "DN",
            "team_name": "team",
            "type": "O",
          },
          "excludeGroups": Array [
            Object {
              "allow_reference": true,
              "create_at": 1,
              "delete_at": 0,
              "description": "",
              "display_name": "DN",
              "has_syncables": false,
              "id": "123",
              "member_count": 3,
              "name": "group_name",
              "remote_id": "",
              "scheme_admin": false,
              "source": "",
              "update_at": 1,
            },
          ],
          "includeGroups": Array [],
          "onAddCallback": [MockFunction],
          "skipCommit": true,
        }
      }
      dialogType={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      disabled={false}
      id="addGroupsToChannelToggle"
      modalId="add_groups_to_channel"
    >
      <Memo(MemoizedFormattedMessage)
        defaultMessage="Add Group"
        id="admin.channel_settings.channel_details.add_group"
      />
    </Memo(Connect(ToggleModalButton))>
  }
  className=""
  id="channel_groups"
  subtitleDefault="Add and remove channel members based on their group membership."
  subtitleId="admin.channel_settings.channel_detail.syncedGroupsDescription"
  titleDefault="Synced Groups"
  titleId="admin.channel_settings.channel_detail.syncedGroupsTitle"
>
  <Connect(GroupList)
    channel={
      Object {
        "group_constrained": false,
        "id": "123",
        "name": "DN",
        "team_name": "team",
        "type": "O",
      }
    }
    groups={
      Array [
        Object {
          "allow_reference": true,
          "create_at": 1,
          "delete_at": 0,
          "description": "",
          "display_name": "DN",
          "has_syncables": false,
          "id": "123",
          "member_count": 3,
          "name": "group_name",
          "remote_id": "",
          "scheme_admin": false,
          "source": "",
          "update_at": 1,
        },
      ]
    }
    isDisabled={false}
    isModeSync={true}
    onGroupRemoved={[MockFunction]}
    setNewGroupRole={[MockFunction]}
    totalGroups={1}
    type="channel"
  />
</AdminPanel>
`;
